src/GrafanaDashboardDeploy/src/Credentials.cs

```diff
@@ -0,0 +1,16 @@
+namespace DotNet.Grafana
+{
+    public class Credentials
+    {
+        /// <summary>
+        /// Credentials!
+        /// </summary>
+        /// <param name="token">Token for bearer auth</param>
+        public Credentials(string token)
+        {
+            Token = token;
+        }
+
+        public string Token { internal get; set; }
```

Comment:
>public string Token { internal get; set; } [](start = 8, length = 42)

Why is the setter public, but the getter internal? It seems like this should just be `public string Token { get; }`

Reply:
My thought was that credentials are secret and should be write-only. Since I wrote this I've seen that read/write is norm.

Reply:
Why is there a setter at all? The constructor accepts one.  In general, types are easier to maintain and debug if they are read only.  Only add a setter if a compelling scenario requires it (and even then I'd consider a immutable "With" type method).

Nothing is more confusing when code looks right, but behaves where because some random method set something on a class you were using.